Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx

```diff
@@ -6,19 +6,25 @@ export default class OfflineIndicator extends React.Component {
   };
 
   componentDidMount() {
-    window.addEventListener('online', this.onlineStatus);
-    window.addEventListener('offline', this.onlineStatus);
+    this.timer = setInterval(() => {
+      fetch('https://www.google.com/generate_204', { mode: 'no-cors' })
+        .then(res => {
+          // with no-cors, res.status is 0 because the javascript ignores all response.
+          // we don't need the response, all we need to know is if the request went through or not
+          if (res.status !== 204) {
```

Comment:
We want to check if the status is 204? `res.status === 204` right?

Reply:
OK, i tested it, then if we know that `res.status` is `0`, we should check for that? `res.status === 0` ?

Reply:
So either check for `res.status === 0` or remove the whole `if` block, since we don't want to check the status